index/index.go

```diff
@@ -935,6 +938,15 @@ func (r *Reader) SortedPostings(p Postings) Postings {
 	return p
 }
 
+// Size returns the size of an index file.
+func (r *Reader) Size() (int64, error) {
+	fInfo, err := os.Stat(r.dir)
+	if err != nil {
+		return -1, err
+	}
+	return fInfo.Size(), nil
```

Comment:
double check the implementation again , but I am pretty sure you don't need all this.

can just simply
`return int64(size), nil`

Reply:
I think you're right. I still think using int64 may be necessary for someone on a 32 bit system that's collecting a lot of metrics. If there's enough chunks they'll exceed the max size of an int, won't they?

Reply:
yes you are right , but I think we can to this only at the end in.
`func blockSize(rr ...SizeReader) (int64, error)`


note to myself: max file size on a 32 bit system is 4gb (4294967296 bytes) , int32 max is 2147483647 , uint32 max is 4294967295

Reply:
I'm not so sure because the majority of the data is in the chunks. If the overflow were to occur, wouldn't it most likely happen in `Size()` of the chunkreader?

Reply:
oh yeah that is true, leave as is than.

Reply:
ping me when ready for another review.

Reply:
probably time to sign the commits as well.